lib/resources/datav2/rest_v2.ts

```diff
@@ -62,30 +65,54 @@ export function dataV2HttpRequest(
     });
 }
 
+function getQueryLimit(
+  totalLimit: number,
+  pageLimit: number,
+  received: number
+): number {
+  let limit = 0;
+  if (pageLimit && pageLimit !== 0) {
+    limit = pageLimit;
+  }
+  if (totalLimit && totalLimit !== 0) {
+    const remaining = totalLimit - received;
+    if (remaining <= 0) {
+      return -1;
+    }
+    if (limit == 0 || limit > remaining) {
+      limit = remaining;
+    }
+  }
+
+  return limit;
+}
+
 export async function* getDataV2(
   endpoint: TYPE,
   path: string,
   options: any,
   config: any
 ): AsyncGenerator<any, void, unknown> {
   let pageToken: string | null = null;
-  let totalItems = 0;
-  const limit = options.limit;
+  let received = 0;
+  const pageLimit = options.pageLimit
+    ? Math.min(options.pageLimit, V2_MAX_LIMIT)
+    : V2_MAX_LIMIT;
+  delete options.pageLimit;
+  const totalLimit = options.limit ?? V2_MAX_LIMIT;
```

Comment:
Maybe I'm missing something here, but this doesn't seems right. If no `limit` is provided by the user, you set it to `V2_MAX_LIMIT`, which is a single page? That basically removes the pagination implementation provided by the SDK.

Reply:
This expression means that we use `options.limit` for the `totalLimit` but if it's null then go with the max value.

Reply:
So how does the user query all trades between start and end?

Reply:
No, the will query with `limit = 10000`.